relay-server/src/actors/processor.rs

```diff
@@ -538,7 +540,14 @@ pub struct EnvelopeProcessorService {
 
 impl EnvelopeProcessorService {
     /// Creates a multi-threaded envelope processor.
-    pub fn new(config: Arc<Config>, _redis: Option<RedisPool>) -> anyhow::Result<Self> {
+    pub fn new(
+        config: Arc<Config>,
+        _redis: Option<RedisPool>,
+        envelope_manager: Addr<EnvelopeManager>,
+        outcome_aggregator: Addr<TrackOutcome>,
+        project_cache: Addr<ProjectCache>,
+        upstream_relay: Addr<UpstreamRelay>,
```

Comment:
nit: We could think about putting these in a separate `Receivers` struct to reduce the number of arguments.

Reply:
yep, this is in the following PR: https://github.com/getsentry/relay/pull/1947